python/GafferUI/NodeGraph.py

```diff
@@ -351,6 +356,18 @@ def __frame( self, nodes, extend = False ) :
 
 		self.__gadgetWidget.getViewportGadget().frame( bound )
 
+	def __nodesHaveUIPosition( self, scriptNode ) :
+
+		for node in scriptNode.children( Gaffer.Node ) :
+			if '__uiPosition' in node :
```

Comment:
This plug is a private implementation detail, not to be relied upon (I actually have plans to change it), as denoted by the __ prefix. I think we need to add a `GraphGadget::hasNodePosition()` method and use that.